### ISSUE TITLE
add `schema-error-property-style` rule

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod-x#32]
Enforce a format or style for error property.

### Options

- `selector`
  - Type: `string`
  - default: `TemplateLiteral,StringLiteral`

### Configuration 

```ts
export default defineConfig({
  // ...
  rules: {
   'zod/schema-error-property-style': ['error', { selector: 'TemplateLiteral' }]
  }
  // ...
})
```

### Valid 

```ts
const stringSchema = z.string().refine(() => true, {
  error: 'asd',
});
```

### Invalid

```ts
const stringSchema = z.string().refine(() => true, {
  error: `asd`,
});
```

```ts
const stringSchema = z.string().refine(() => true, {
  error: () => 'asd',
});
```

```ts
const stringSchema = z.string().refine(() => true, {
  error() {
    return 'asd';
  },
});
```

```ts
const stringSchema = z.string().refine(() => true, {
  error: () => {
    return 'asd';
  },
});
```
